### PR TITLE
Update test timeout and add wait_for_glbc_deletion for ingress resources cleanup.

### PR DESCRIPTION
**Pull request by sawsa307**
* Add wait_for_glbc_deletion to wait for all ingress resources cleaned up before deleting cluster.
* Update test timeout so test won't keep hanging.

### CODE REVIEW

File: test/helper.sh [hunk 1]
```diff
@@ -98,6 +98,76 @@ wait_for_managed_cert() {
...
+# Arguments:
+#   Name of the ingress.
+#   Namespace of the ingress.
+# Returns:
+#   0 if resources are deleted within 1 hour, 1 if not.
+wait_for_glbc_deletion() {
+    local ingress_name test_name ns ing_resource_name
+    ingress_name="$1"
+    test_name="$2"
+    resource_suffix=$(get_hash "${test_name}")
```
Comment (sawsa307) [inline test/helper.sh:111]: Make it local

File: test/helper.sh [hunk 2]
```diff
@@ -98,6 +98,76 @@ wait_for_managed_cert() {
...
+    network="gke-net-recipes-${resource_suffix}"
```
Comment (sawsa307) [inline test/helper.sh:112]: Same applies.

File: test/helper.sh [hunk 3]
```diff
@@ -98,6 +98,76 @@ wait_for_managed_cert() {
...
+
+    ns="${test_name}"
+    ing_resource_name="${ns}-${ingress_name}"
+    for ATTEMPT in $(seq 360); do
+        local fwr thp thsp um bs neg resource_suffix network
```
Comment (sawsa307) [inline test/helper.sh:117]: Remove resource_suffix network

File: Makefile [hunk 4]
```diff
@@ -29,8 +29,8 @@ test: bin/recipes-test
 	bin/recipes-test \
 		--run-in-prow=$(RUN_IN_PROW) \
 		--boskos-resource-type=$(BOSKOS_RESOURCE_TYPE) \
-		-test.v
-		-test.timeout=0
+		-test.v \
+		-test.timeout=180m
```
Comment (bowei) [inline Makefile:33]: Does the test really take 3 hours?

**Review by bowei — APPROVED** [review]

**Pull request merged**